packages/frontend/src/components/UserEvent.jsx

```diff
@@ -3,12 +3,14 @@ import { Link } from "react-router-dom";
 import './eventCard.css'
 
 
-const UserEvent = ({ repReceived, attester, epoch })  => {
+const UserEvent = ({ attester, epoch, time })  => {
```

Comment:
I've changed these components (`UnirepEvent` above as well) to align with the design updates.

[This](https://github.com/Unirep/explorer/pull/3#discussion_r1061679442) comment from UI PR last week:

> This is kinda hard to explain because we don't have a lot of data yet, but the idea behind the stores is that all data lives there and we don't pass things like `repReceived`, `epoch`, `attester` to components.
> 
> Instead each event (and every other object) has an identifier, and we load the object from the store using the identifier. That way we have a single source of truth, and we can change that source and change the ui in all places. Basically we get reactive elements for free and avoid state conflicts that arise by passing values around.
> 
> Then we access `repReceived`, `epoch`, `attester` on the object that we loaded from the store.

yes, makes more sense now that I am receiving data on the FE.. but not quite. Like I see that it's extra steps to pass the data accessed from the store to the component just to get it back again, but I don't see how the component can access the data from the store otherwise.

Does this mean that using React components is altogether unnecessary here, and I should be able to iterate through the data objects directly on Home/Attester/User pages to display the events?

Reply:
I think it will make more sense when you change the context structure. Generally you should use a component when there's some re-usable logic, or just logic you want to isolate.